mlflow_faculty/artifacts.py

```diff
@@ -50,37 +41,70 @@ def __init__(self, artifact_uri):
                 )
             )
 
-        cleaned_path = parsed_uri.path.strip("/")
-        first_part = cleaned_path.split("/")[0]
+        cleaned_path = parsed_uri.path.strip("/") + "/"
+        first_part, remainder = cleaned_path.split("/", 1)
+
         try:
-            project_id = UUID(first_part)
+            self.project_id = UUID(first_part)
         except ValueError:
             raise ValueError(
                 "{} in given URI {} is not a valid UUID".format(
                     first_part, artifact_uri
                 )
             )
 
-        self._session = session.get()
-        s3_bucket = self._session.bucket(project_id)
-        s3_uri = "s3://{}/{}".format(s3_bucket, cleaned_path)
-        s3_client_factory = partial(self._session.s3_client, project_id)
+        self.datasets_artifact_root = "/" + remainder
 
-        self._s3_repository = _S3ArtifactRepositoryWithClientOverride(
-            s3_uri, s3_client_factory
+    def _datasets_path(self, artifact_path):
+        return posixpath.normpath(
+            posixpath.join(
+                self.datasets_artifact_root, artifact_path.lstrip("/")
+            )
         )
 
-    def get_path_module(self):
-        return self._s3_repository.get_path_module()
-
     def log_artifact(self, local_file, artifact_path=None):
-        return self._s3_repository.log_artifact(local_file, artifact_path)
+        if artifact_path is None:
+            artifact_path = os.path.basename(local_file)
+        datasets_path = self._datasets_path(artifact_path)
+        datasets.put(local_file, datasets_path, self.project_id)
 
     def log_artifacts(self, local_dir, artifact_path=None):
-        return self._s3_repository.log_artifacts(local_dir, artifact_path)
+        if artifact_path is None:
+            artifact_path = "./"
+        datasets_path = self._datasets_path(artifact_path)
+        datasets.put(local_dir, datasets_path, self.project_id)
+
+    def list_artifacts(self, path=None):
+        if path is None:
+            path = "./"
+        datasets_path = self._datasets_path(path)
+
+        # Make sure path interpreted as a directory
+        prefix = datasets_path.rstrip("/") + "/"
+
+        # Go directly to the object store so we can get file sizes in the
+        # response
+        client = faculty.client("object")
+
+        list_response = client.list(self.project_id, prefix)
```

Comment:
Maybe I am missing something, but can't we just use `datasets.ls()` to save us having to deal with pagination?

Reply:
`datasets.ls()` returns just the paths of the objects - we also need the sizes here.